### ISSUE TITLE
'gnt-cluster upgrade' should automatically detect a previous failed upgrade

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#962]
Originally reported of Google Code with ID 906.

```
Feature request:

It would be great if the "automatized" upgrade command would detect a previously failed upgrade, and switch to its --resume mode if the version being upgraded to is the same.

A small issue that might arise with this is the situation where a resume is in progress and another 'gnt-cluster upgrade' command is issued.  I guess this could be mitigated with a lockfile to ensure only one upgrade process is active at any one time.

Thanks a lot!
```
Originally added on 2014-07-31 15:28:41 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2014-08-01 07:14:44 +0000 UTC.

Added Labels: Type-Defect Priority-High
Changed State: Accepted
Added to Milestone: Release2.10

**Comment by Ganeti-Issues-Migrator** [comment]
```
Assigning this to aehlig as this is a bigger change to the control flow.
```
Originally added on 2014-08-05 09:08:27 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
commit de23298f39ec2c63291b1009eb771d3b86043a26
Author: Klaus Aehlig <aehlig@google.com>
Date:   Wed Aug 13 14:08:22 2014 +0200

    On upgrades, check for upgrades to resume first
    
    Make gnt-cluster upgrade refuse to upgrade if an upgrade to be resumed
    is present. For the convenience of the user, consider an upgrade
    command to the same target version as the upgrade to be resumed as an
    equivalent way of saying that the pending upgrade is to be resumed.
    Fixes issue 906.
    
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Helga Velroyen <helgav@google.com>

```
Originally added on 2014-08-18 11:12:10 +0000 UTC.

Mentioned Issue with GoogleCode ID 906 has been migrated to GitHub with ID #962.
Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
Released as part of 2.11.6
```
Originally added on 2014-09-22 15:56:40 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Released as part of 2.11.6
```
Originally added on 2014-09-22 15:57:10 +0000 UTC.

Changed State: Released
